### PR TITLE
Python: Fixes for D205 issues

### PR DESCRIPTION
**Pull request by mderoy**
https://github.com/apache/iceberg/issues/7776

### CODE REVIEW

File: python/pyiceberg/conversions.py [hunk 1]
```diff
@@ -109,6 +109,8 @@ def _(primitive_type: BooleanType, value_str: str) -> Union[int, float, str, uui
 @handle_none
 def _(primitive_type: PrimitiveType, value_str: str) -> int:
     """
+    Converts a string to an integer value
```
Comment (Fokko) [inline python/pyiceberg/conversions.py:112]: Shouldn't this one be one line up?

**Review by Fokko — APPROVED** [review]
One minor comment, looking great @mderoy! You're on fire!

File: python/pyiceberg/conversions.py (hunk 1 above)
Comment (mderoy) [reply python/pyiceberg/conversions.py]: whoops :) fixed with https://github.com/apache/iceberg/pull/7823/commits/d450891f870e94d59022ca0dbec79e1e667ba4fa

**Comment by Fokko** [comment]
@mderoy Could you rebase? https://github.com/apache/iceberg/pull/7829 went just in and it looks it has a lot of conflicts

File: python/pyiceberg/utils/deprecated.py [hunk 2]
```diff
@@ -20,9 +20,9 @@
 
 
 def deprecated(deprecated_in: str, removed_in: str, help_message: Optional[str] = None) -> Callable:  # type: ignore
-    """
-    This is a decorator which can be used to mark functions as deprecated.
-    It will result in a warning being emitted when the function is used.
+    """A decorator which can be used to mark functions as depricated.
```
Comment (Fokko) [inline python/pyiceberg/utils/deprecated.py:23]:
```suggestion
    """A decorator which can be used to mark functions as deprecated.
```

**Review by Fokko — APPROVED** [review]
Looking great, thanks again @mderoy

**Pull request merged**